adobe/spectrum-web-components
# feat(combobox): add `pending` state

## Description

Adds `pending` property for `sp-combobox`, enabling the pending state.

[Spectrum CSS](https://opensource.adobe.com/spectrum-css/combobox.html)
![Screenshot 2024-05-16 at 12 10 02](https://github.com/adobe/spectrum-web-components/assets/13311865/1bb887cc-2fb2-495d-b2a3-02080e055a39)

## Related issue(s)



- https://github.com/adobe/spectrum-web-components/issues/4382

## Motivation and context


A combo box can indicate that content is loading if system processes delay the display of the combo box content. 

## How has this been tested?




    1. Go [here](https://rocss-pending-combobox--spectrum-web-components.netlify.app/components/combobox/)
    2. Observe documentation entries for `quiet` and `pending` states are now available
    1. Go [here](https://rocss-pending-combobox--spectrum-web-components.netlify.app/storybook/?path=/story/combobox--pending)
    2. Combobox should receive focus when in pending state
    3. Your are able to type inside it, however, it does not open the suggestion list while it is pending
    4. Toggling the `open` control does not open the suggestion list
    1. Go [here](https://rocss-pending-combobox--spectrum-web-components.netlify.app/storybook/?path=/story/combobox--pending)
    2. Inspect the combobox using Screen Reader (I used VoiceOver on Mac)
    3. It reads the "Pending" label together with the Combobox's visible label.
    1. Go [here](https://rocss-pending-combobox--spectrum-web-components.netlify.app/storybook/?path=/story/combobox--pending)
    2. Toggle between LTR / RTL
    3. The progress circle is correctly positioned near the picker-button
    1. Go [here](https://rocss-pending-combobox--spectrum-web-components.netlify.app/storybook/?path=/story/combobox--pending)
    2. Enable `disabled` control -> all looks good
-   [ ] _Test case 6_
    1. Go [here](https://rocss-pending-combobox--spectrum-web-components.netlify.app/storybook/?path=/story/combobox--pending)
    2. Enable `invalid` control  -> progress circle takes precedence over the alert icon
-   [ ] _Test case 7_
    1. Go [here](https://rocss-pending-combobox--spectrum-web-components.netlify.app/storybook/?path=/story/combobox--pending)
    2. Enable `quiet` control -> all looks good


## Screenshots (if appropriate)

## Types of changes




## Checklist




## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.
